### PR TITLE
Fix #728: PyTorchDataGenerator always returns the same batch

### LINKED ISSUE #728: PyTorchDataGenerator always returns the same batch
**Describe the bug**
`art.data_generators.PyTorchDataGenerator` returns the same batch each time its `get_batch()` method is called. Judging by how it is used (e.g. in `art.defences.trainer.AdversarialTrainer`, it should return a different batch every time.

**To Reproduce**
Try the following code snippet:
```python
import torch
import numpy as np
from torchvision.datasets import CIFAR10
import torchvision.transforms as transforms
from torch.utils.data import DataLoader
from art.data_generators import PyTorchDataGenerator

# Using CIFAR-10, feel free to replace this with any dataset of your choice if you don't want to download it
cifar = CIFAR10("./cifar", download=True, train=True, transform=transforms.ToTensor())  
dataloader = DataLoader(cifar, batch_size=32, shuffle=False)
train_generator = PyTorchDataGenerator(dataloader, size=len(cifar), batch_size=32)

batch1_x, batch1_y = train_generator.get_batch()
batch2_x, batch2_y = train_generator.get_batch()

print(np.array_equal(batch1_x, batch2_x), np.array_equal(batch1_x, batch2_x))
```
It prints `True True`, whereas it should print `False False`.

**Expected behavior**
As mentioned, I would expect `get_batch()` to return a different batch each time it is called. For now, I am using the following class as a workaround:
```python
class CustomDataGenerator(DataGenerator):
    """
    Wrapper class on top of the PyTorch native data loader :class:`torch.utils.data.DataLoader`.
    """

    def __init__(self, iterator: "torch.utils.data.DataLoader", size: int, batch_size: int) -> None:
        """
        Create a data generator wrapper on top of a PyTorch :class:`DataLoader`.
        :param iterator: A PyTorch data generator.
        :param size: Total size of the dataset.
        :param batch_size: Size of the minibatches.
        """
        from torch.utils.data import DataLoader

        super().__init__(size=size, batch_size=batch_size)
        if not isinstance(iterator, DataLoader):
            raise TypeError("Expected instance of PyTorch `DataLoader, received %s instead.`" % str(type(iterator)))

        self._iterator: DataLoader = iterator
        self.current = iter(self.iterator)

    def get_batch(self) -> tuple:
        """
        Provide the next batch for training in the form of a tuple `(x, y)`. The generator should loop over the data
        indefinitely.
        :return: A tuple containing a batch of data `(x, y)`.
        :rtype: `tuple`
        """
        try:
            batch = list(next(self.current))
        except StopIteration:
            self.current = iter(self.iterator)
            batch = list(next(self.current))

        for i, item in enumerate(batch):
            batch[i] = item.data.cpu().numpy()

        return tuple(batch)
```

**Screenshots**
N/A

**System information (please complete the following information):**
 - Ubuntu 18.04
 - Python 3.7.4
 - ART 1.4.2
 - PyTorch 1.5.1

### PR DESCRIPTION
**Pull request by hornel**
Signed-off-by: hornel <hornel@ethz.ch>

# Description

Fixes `art.data_generators.PyTorchDataGenerator` and `art.data_generators.MXDataGenerator`. Previously, they always returned the same batch (assuming the underlying DataLoader had shuffle disabled), whereas they now properly iterate through the underlying DataLoader.

Fixes #728. I also noticed the same issue with MXDataGenerator and provided the same fix.

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] All unit tests in `tests/test_data_generators/TestPyTorchGenerator`
- [x] All unit tests in `tests/test_data_generators/TestMXGenerator`

**Test Configuration**:
- Ubuntu 18.04
- Python 3.7.4
- ART 1.4.2
- PyTorch 1.5.1 / MXNet 1.6.0

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]

**Comment by beat-buesser** [comment]
@hornel Thank you very much!
